### ISSUE TITLE
Parse Cloudfront logs for add-on user stats

### ISSUE THREAD

**Issue by Sancus** [thundernest/addons-server#4]
In front of versioncheck.addons.thunderbird.net, Cloudfront sits. This means that most http requests won't reach the actual server, so we need to parse these logs to figure out user pings for add-on stats.

**Comment by ovari** [comment]
Are you able to please advise when the statistics will be once again available for add-ons? Thank you

**Comment by ovari** [comment]
**[Thunderbird/StatusMeetings/2018-09-18](https://wiki.mozilla.org/Thunderbird/StatusMeetings/2018-09-18)**
**[sancus](https://wiki.mozilla.org/Thunderbird/StatusMeetings/2018-09-18#sancus)**
* Still working on addons-server update and stats fixes, should be done this week.

Are you able to please advise when the statistics will be once again available for add-ons? Thank you

**Comment by Sancus** [comment]
I don't have an ETA for you but this hasn't been forgotten and I am actively working on it.

**Comment by RealRaven2000** [comment]
thanks this is badly needed. We (add-on devs) have to make a lot of hard decisions for the future whene we commit a maxver for the 60.3 release. A lot of it is "guess what's going away?".I have already been cut by Postbox deciding to drop 3rd party addons in 5.52b6

**Comment by RealRaven2000** [comment]
do you need any info from us to help closing this issue?

**Comment by CardBook** [comment]
any news ?

**Comment by Sancus** [comment]
https://addons.thunderbird.net/en-US/thunderbird/addon/lightning/statistics/usage/?last=90

ADU stats should now be available, including data back to July 20th, the first full day that the site was up.

This does not include downloads because those rely on a different server, which currently has Cloudflare in front of it and so we don't have access to logs. I plan to change this eventually but it will be a couple of months probably.

There's a similar situation with personas/lightweight themes, which ping a completely different server that we currently aren't running. Updates on these seemed pretty low value, since they rarely get updated and can't really "break". I do have a plan to put these on the same servers as versioncheck(which is different from the way AMO runs them), but again not super high priority.

Complete Themes like [TT DeepDark](https://addons.thunderbird.net/en-US/thunderbird/addon/tt-deepdark/) use the regular addon update server and so they do have ADUs tracked.

**Comment by ovari** [comment]
> ADU stats should now be available, including data back to July 20th, the first full day that the site was up.

Thank you

**Comment by RealRaven2000** [comment]
Hurray! Thanks for this valuable tool. Is it possible to run three statistics on Thunderbird itself? It would be interesting to see how add-on growth and Application correlate.

**Comment by mikehardy** [comment]
Only a guess on my part but I've noticed that they post at irregular times, and there was a bug lag over the winter holidays so I'm guessing they are not fully automated since they are newly revived and there is some sort of staffing thing going on. Curious if I am right? If so I obviously want fresh stats but I also hope everyone's alright. Cheers.

**Comment by Sancus** [comment]
The cron job seems to have gotten stuck for some reason, I unstuck it and will keep an eye on it.

**Comment by mikehardy** [comment]
Not urgent - but if it is indeed a cronjob, I expect it's daily? And it appears it is stuck again, but only just now. Maybe since it's recent the conditions causing it to have issues are still present.

**Comment by mikehardy** [comment]
@Sancus I understand these are not urgent, I'm only raising a comment here in case you weren't aware and you want to be - it appears the updates stalled again this week? Thanks for the work here :-)

**Comment by ovari** [comment]
@Sancus I understand these are not urgent, I'm only raising a comment here in case you weren't aware and you want to be - it appears the updates stalled again this week? Thanks for the work here :-)

**Comment by mikehardy** [comment]
I looked and I think the 2-day lag is normal, it looks like it's caught up again since I posted a few days ago and has the normal data I'd expect. And my little plugin has a very slowly-but-steadily growing audience which is why I like to watch ;-)

**Comment by Sancus** [comment]
I rewrote stats stuff so that it all runs on one machine with every part triggered by one script now. It was flaky before because there were several machines in different places involved. The stats script runs at 12pm PT now for the day before(so, 1 hour before this comment, it ran for March 28th) and stats should be updated by 12:15 in the worst case.

I'm going to close this, as it's gotten a bit ridiculously long and the actual bug was solved a long time ago. I opened #80 for download stats tracking just so it isn't completely forgotten, but I'm not planning to work on it any time soon.
